src/main/java/org/broadinstitute/hellbender/utils/test/SamAssertionUtils.java

```diff
@@ -363,6 +371,44 @@ public static void assertEqualBamFiles(final File resultFile, final File expecte
         }
     }
 
+    // validate that this file has CRAM contents (no matter what the extension)
+    protected static boolean isCRAMFile(final File putativeCRAMFile, final File referenceFile) {
+        boolean isCRAMFile = false;
+        CRAMFileReader cramReader = null;
+        try {
+            // bypass SAMReaderFactory, which will hand back a reader for practically anything
+            cramReader = new CRAMFileReader(
+                    putativeCRAMFile,
+                    new htsjdk.samtools.cram.ref.ReferenceSource(referenceFile)
+            );
+            isCRAMFile = true;
+        }
+        catch (RuntimeException e) {
+            if (!e.getMessage().contains("Unknown file format")) {
+                throw e;
```

Comment:
Parsing `getMessage()` like this is pretty gross/brittle -- can you either find another way to detect CRAM (using a more specific/low-level method/class in htsjdk) or open an htsjdk ticket to use a separate exception subclass for `Unknown file format` errors.